### ISSUE TITLE
Linter plugins: Support RegExps and other unserializable values in rule options

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#20530]
#20264 revealed a bug with JS plugins.

`eslint-plugin-playwright` accepts `RegExp`s as options. e.g.:

```js
config = {
  rules: {
    "playwright/valid-test-tags": [
      "error",
      { disallowedTags: ['@skip', /^@temp-/] },
    ],
  },
};
```

If Oxlint config file is JSON, then there's no way to provide a `RegExp` as a rule option, because `RegExp`s are not valid in JSON. But now we have `oxlint.config.ts`, it *is* possible.

Only solutions I can see are:

1. Use a different serialization format than JSON for sending config back/forwards between Rust.
2. Keep a copy of the config object on JS side and Rust sends back a reference to where to find the options object in the config (e.g. `rules["playwright/valid-test-tags"]`).

For option 1, we'd need a serialization lib which produces valid JSON so we can continue to use `serde` on Rust side. e.g. a `RegExp` is encoded as `{ "$$__REGEX__hTj78X__$$": true, "pattern": "^@temp-", "flags": "" }`. I'm not sure if such a thing exists or if we'd have to build it ourselves. Something like [serialize-javascript](https://npmx.dev/package/serialize-javascript) unfortunately would not work as it doesn't produce valid JSON.

Option 2 is in some ways more complicated, but more robust - it would support even JS constructs as rule options which are impossible to serialize, e.g. functions. I've not encountered a rule which accepts functions as options yet, but I'd bet my bottom dollar that one exists!

Actually there is one serializer that fully supports functions, including properly handling vars captured from function scope - my own [livepack](https://npmx.dev/package/livepack). But to be honest I wouldn't recommend using it!

**Comment by nebrius** [comment]
Just came to the ask for the same thing, which is needed for my plugin at https://github.com/nebrius/eslint-plugin-fast-import (regex's for paths).

**Comment by absidue** [comment]
The way that eslint-plugin-vue handles it is by allowing you to write JavaScript regex literals inside of strings maybe oxlint could do something similar for the JSON config e.g.: https://eslint.vuejs.org/rules/prop-name-casing.html#ignoreprops-foo-bar-a-z-u
